### DISCUSSION TITLE
I got an idea today about multi-color printing, that could be a game changer for the slicer that will implement it and make purge tower obsolete :)

### DISCUSSION THREAD

**Discussion by DzzD** [supermerill/SuperSlicer#1915] [General]
Dont know if someone already think of it but here is my idea :

Using a purge bucket greatly reduce the amount of filament used and print time on multicolor printer using single nozzle and multiple extruders as the filament purge is only done when required, so for an object of 100 layers if there is only the top layer with a different color, the filament will only do one purge (into the purge bucket), without purge bucket the printer will build a "purge tower" of 100 layers with a lot of filament and time wasted.

Here is the (two) solution I propose to enable purge bucket on any FFF printer :

1 - Build a purge bucket on the bed before starting the model print, and use it as a regular purge bucket with custm tool change G-code.
2 - Build the purge bucket layer by layer while printing (it will also be the ooze shield) will take more time but a lot easier to implement

![image](https://user-images.githubusercontent.com/12198339/142387453-6bdff78c-24ad-43d4-bdc2-2a02f39345bc.png)

Note that the purge bucket can be  any shape like very thin rectangle along the X o Y axes, or rounded, etc... also height does not have to be the same as the model

Edit: Plz dont tell me it is the same as a purge block, in this solution we draw only one perimeter/layer and sometime purge a little filament while a purge block is a solid infilled model

**Comment by DzzD** [comment]
Note that with new placeholder for custom g-code current_pos_x & current_pos_y, I think the second solution can already be done easily by custom gcode

**Reply by supermerill** [reply]
> Edit: Plz dont tell me it is the same as a purge block, in this solution we draw only one perimeter/layer and sometime purge a little filament while a purge block is a solid infilled model

it's the same a a purge block, as it only draw the perimeter when there is nothing to purge.

**Reply by DzzD** [reply]
So it is a purge perimeter, not a purge block

**Reply by DzzD** [reply]
For model with few color change it will not use more filament then a ooze shield does

**Reply by supermerill** [reply]
I don't see the diff with the current purge tower.

**Reply by DzzD** [reply]
the main difference is to use an empty block rather than a solid one, probably faster and using less filament in a lot of case

**Reply by DzzD** [reply]
inded It is interresting only for model that have not many color on each layer

**Reply by DzzD** [reply]
But yes you are right, if we can have purgeblock that grow only when  color change this is no more interresting

**Reply by supermerill** [reply]
The current purge tower isn't solid, unless you have the max tool change in the layer.

**Comment by DzzD** [comment]
Ok that's not a good idea :) but what about a purgeblock that grow only when color change ?

Here is for me the problem : 
![image](https://user-images.githubusercontent.com/12198339/142394014-793d9984-9350-44e7-9494-2ba286cab690.png)

In this case the burge block should be 4 layers height

**Reply by supermerill** [reply]
did you look inside the tower?

**Reply by DzzD** [reply]
Oups no, I have seen it now... so finally that's not so bad

**Reply by DzzD** [reply]
Sorry I did not know this was already possible to do a smaller purgeblock too...

**Comment by DzzD** [comment]
there are mornings when we wake up thinking we have a great idea and in fact it sucks :)

**Comment by neophyl** [comment]
If you are interested in purge buckets there is an (older) thread on the Prusa forum detailing some mods on a MK3. 
 As the software they were using to slice with was Prusa Slicer almost everything would probably still be applicable to Susi to. 
 
https://forum.prusaprinters.org/forum/original-prusa-i3-mmu2s-mmu2-general-discussion-announcements-and-releases/the-mmu2-purge-bucket-thread-mod-wip/#post-156528

**Reply by DzzD** [reply]
Thanks, no not really interrested in "real" purge bucket, I already  got one wich works well, I was just thinking of an idea to make it possible on any printer, but that was just a bad idea, I personaly dont need it (I already did a g-code srcipt for simplify3d & Susi that works well on my printer).

**Comment by Kadah** [comment]
The "no sparse layers" wipe tower option will save even more filament at the cost of build volume on printers/jobs that can support it. The purge block will only get used when there is a tool change and is skipped for layers that don't, The height different is accounted for by lowering the print head back down to the tower.

So on that example traffic cone, the purge block would be 4-5 layers total.

The downside is the possible additional time due to the z moves and that not every printer can do this for every print. An i3 style printer will not have an issue doing this at the expense of the whole backside of the bed at the depth of the print head. The time lost-gain will depend on the printer and the job; z moves on i3 printers are slow but its also not having to print sparse on the tower every layer.

When I have the clearance, I use this option by default.

**Comment by Key-Alex** [comment]
Isn't it possible to just start printing a different color of plastic inside the model when printing an infill, then by the time the part of the part consisting of plastic of a different color is printed, it will already change, and the entire color change will occur inside the model? It's just a question.

**Reply by supermerill** [reply]
You already can wipe into infill
https://help.prusa3d.com/en/article/wipe-tower_125010#wipe-to-infill
